src/group_theory/p_group.lean

```diff
@@ -280,4 +293,60 @@ begin
   simpa [this] using hn₁,
 end
 
+section p2comm
+
+variables [fintype G] [fact p.prime] {n : ℕ} (hGpn : card G = p ^ n)
+include hGpn
+open subgroup
+
+/-- The cardinality of the `center` of a `p`-group is `p ^ k` where `k` is positive. -/
+lemma card_center_eq_prime_pow (hn : 0 < n) [fintype (center G)] :
+  ∃ k > 0, card (center G) = p ^ k :=
+begin
+  have hcG := to_subgroup (of_card hGpn) (center G),
+  rcases iff_card.1 hcG with ⟨k, hk⟩,
+  haveI : nontrivial G := (nontrivial_iff_card $ of_card hGpn).2 ⟨n, hn, hGpn⟩,
+  have := center_nontrivial (of_card hGpn),
+  rwa nontrivial_iff_card hcG at this
+end
+
+omit hGpn
+
+/-- The quotient by the center of a group of cardinality `p ^ 2` is cyclic. -/
+lemma cyclic_center_quotient_of_card_eq_prime_sqr (hG : card G = p ^ 2) :
+  is_cyclic (G ⧸ (center G)) :=
+begin
+  classical,
+  rcases card_center_eq_prime_pow hG zero_lt_two with ⟨k, hk0, hk⟩,
+  have hk2 : k ≤ 2, from (nat.pow_dvd_pow_iff_le_right (fact.out p.prime).one_lt).1
+    (hG ▸ hk ▸ card_subgroup_dvd_card (center G)),
+  rw [card_eq_card_quotient_mul_card_subgroup (center G), hk] at hG,
+  rcases k with  _ | _ | _ | k,
+  { exact (lt_irrefl _ hk0).elim },
+  { rw [pow_two, pow_one, nat.mul_left_inj (fact.out p.prime).pos] at hG,
+    exact is_cyclic_of_prime_card hG },
+  { conv_rhs at hG { rw ← one_mul (p ^ 2) },
+    rw [nat.mul_left_inj (pow_pos (fact.out p.prime).pos _)] at hG,
+    exact @is_cyclic_of_subsingleton _ _ ⟨fintype.card_le_one_iff.1 (le_of_eq hG)⟩ },
+  { simpa [nat.succ_le_succ_iff] using hk2 }
+end
+
+/-- A group of order `p ^ 2` is commutative. See also `comm_group_of_card_eq_prime_sqr` for the
+`comm_group` instance. -/
+lemma commutative_of_card_eq_prime_sqr (hG : card G = p ^ 2) : ∀ a b : G, a * b = b * a :=
+begin
+  classical,
+  by haveI : is_cyclic (G ⧸ (center G)) :=
+    cyclic_center_quotient_of_card_eq_prime_sqr hG;
+  exact commutative_of_cyclic_center_quotient (quotient_group.mk' (center G)) (by simp)
+end
+
+/-- A group of order `p ^ 2` is commutative. See also `commutative_of_card_eq_prime_sqr`
+for just the proof that `∀ a b, a * b = b * a` -/
+def comm_group_of_card_eq_prime_sqr (hG : card G = p ^ 2) : comm_group G :=
```

Comment:
Have you seen the def `comm_group_of_cycle_center_quotient` in `group_theory/specific_groups/cyclic.lean`?

Reply:
I don't think this duplication would cause any problem.